libs/@hashintel/petrinaut/src/state/editor-store.ts

```diff
@@ -59,152 +46,99 @@ export type EditorState = {
-            return "arc";
-          }
-
-          // Check places
-          if (sdcpn.places.some((place) => place.id === id)) {
-            return "place";
-          }
-
-          // Check transitions
-          if (sdcpn.transitions.some((transition) => transition.id === id)) {
-            return "transition";
-          }
-
-          // Check types
-          if (sdcpn.types.some((type) => type.id === id)) {
-            return "type";
-          }
-
-          // Check differential equations
-          if (
-            sdcpn.differentialEquations.some((equation) => equation.id === id)
-          ) {
-            return "differentialEquation";
-          }
-
-          // Check parameters
-          if (sdcpn.parameters.some((parameter) => parameter.id === id)) {
-            return "parameter";
-          }
-
-          return null;
-        },
-
-        // Dragging state
-        draggingStateByNodeId: {},
-        setDraggingStateByNodeId: (state) =>
-          set(
-            { draggingStateByNodeId: state },
-            false,
-            "setDraggingStateByNodeId",
-          ),
-        updateDraggingStateByNodeId: (updater) =>
-          set(
-            (state) => ({
-              draggingStateByNodeId: updater(state.draggingStateByNodeId),
+      (set) =>
+        ({
+          globalMode: "edit",
+          setGlobalMode: (mode) =>
+            set({ globalMode: mode }, false, { type: "setGlobalMode", mode }),
+
+          editionMode: "select",
+          setEditionMode: (mode) =>
+            set({ editionMode: mode }, false, { type: "setEditionMode", mode }),
+
+          // UI state
+          isLeftSidebarOpen: true,
+          setLeftSidebarOpen: (isOpen) =>
+            set({ isLeftSidebarOpen: isOpen }, false, {
+              type: "setLeftSidebarOpen",
+              isOpen,
+            }),
+
+          // Selected Resource ID
+          selectedResourceId: null,
+          setSelectedResourceId: (id) =>
+            set({ selectedResourceId: id }, false, {
+              type: "setSelectedResourceId",
+              id,
+            }),
+
+          // Selection
+          selectedItemIds: new Set(),
+          setSelectedItemIds: (ids) =>
+            set({ selectedItemIds: ids }, false, {
+              type: "setSelectedItemIds",
+              ids,
             }),
-            false,
-            "updateDraggingStateByNodeId",
-          ),
-        resetDraggingState: () =>
-          set({ draggingStateByNodeId: {} }, false, "resetDraggingState"),
-
-        __reinitialize: () => {
-          set(
-            {
-              globalMode: "edit",
-              editionMode: "select",
-              isLeftSidebarOpen: true,
-              selectedResourceId: null,
-              selectedItemIds: new Set(),
-              draggingStateByNodeId: {},
-            },
-            false,
-            { type: "initializeEditorStore" },
-          );
-        },
-      }),
```

Comment:
For some reason the type checking in `create<>` isn't showing a compiler error if a function in the type isn't provided in the create call... this does catch it. Not sure what's going on there.